help-docs: Update search for messages article about "group-pm-with".

laurynmm opened the pull request:
Updates the help center article on searching for messages to have the correct format for the "group-pm-with" operator.
Also, reorders the list so that the private message searches are all together.

See [relevant CZO conversation](https://chat.zulip.org/#narrow/stream/19-documentation/topic/.22group-pm-with.22.20search.20with.20multiple.20users/near/1456141).

---

**Notes**:
- Would it make sense to add a section after **Excluding messages** for combining operators to cover the use case of using the "group-pm-with" operator multiple times to search for group private messages with more than one person, but not exclusively with those specific users?

- I moved the `streams:public` operator to the end of the list because I felt like it stands out more there.

---

**Screenshots and screen captures:**

<details>
<summary>Updated full list of operators</summary>

[Current documentation](https://zulip.com/help/search-for-messages#list-of-operators)
![Screenshot from 2022-10-28 19-38-03](https://user-images.githubusercontent.com/63245456/198700067-3e6609ac-25f5-4877-ad83-bfe228993261.png)
</details>

---

**Self-review checklist**

<!-- Prior to submitting a PR, follow our step-by-step guide to review your own code:
https://zulip.readthedocs.io/en/latest/contributing/code-reviewing.html#how-to-review-code -->

<!-- Once you create the PR, check off all the steps below that you have completed.
If any of these steps are not relevant or you have not completed, leave them unchecked.-->

- [x] [Self-reviewed](https://zulip.readthedocs.io/en/latest/contributing/code-reviewing.html#how-to-review-code) the changes for clarity and maintainability
      (variable names, code reuse, readability, etc.).

Communicate decisions, questions, and potential concerns.

- [x] Explains differences from previous plans (e.g., issue description).
- [ ] Highlights technical choices and bugs encountered.
- [x] Calls out remaining decisions and concerns.
- [ ] Automated tests verify logic where appropriate.

Individual commits are ready for review (see [commit discipline](https://zulip.readthedocs.io/en/latest/contributing/version-control.html)).

- [x] Each commit is a coherent idea.
- [x] Commit message(s) explain reasoning and motivation for changes.

Completed manual review and testing of the following:

- [x] Visual appearance of the changes.
- [ ] Responsiveness and internationalization.
- [ ] Strings and tooltips.
- [ ] End-to-end functionality of buttons, interactions and flows.
- [ ] Corner cases, error conditions, and easily imagined bugs.

alya commented:
The screenshot looks good to me.

What do you think about adding something like this to `is:public`, to clarify why you'd want to use it?

> ... in the organization, not only streams you are subscribed to.

alya commented:
I was also considering whether we should add explicit category labels to the list, since it's currently _implicitly_ organized by category. But I wasn't sure I could come up with helpful labels for all the groupings.

laurynmm commented:
**Updates**:
- Added [this clarification](https://github.com/zulip/zulip/pull/23372#issuecomment-1295410989) for `streams:public` operator description. This is in both commits.
- Added a second commit that does some larger revisions on the page.

> I was also considering whether we should add explicit category labels to the list, since it's currently _implicitly_ organized by category. But I wasn't sure I could come up with helpful labels for all the groupings.

I had a similar thought, but when I started adding headers, I felt like the whole page needed to be adjusted a bit :sweat_smile:. I put those changes in a second commit, so it's easy enough to drop. I'm putting in screenshots of the page for each commit below.

Also I thought that @alya  might have some thoughts on how to clean-up or adjust that second commit to be a bit more what we're looking for. Or to address [this idea](https://chat.zulip.org/#narrow/stream/19-documentation/topic/.22group-pm-with.22.20search.20with.20multiple.20users/near/1456523) that came up on CZO.

**Screenshots**:
[Current documentation](https://zulip.com/help/search-for-messages)

<details>
<summary>Commit 1 - fixes the error and makes small change to list order</summary>

![Screenshot from 2022-10-31 17-40-32](https://user-images.githubusercontent.com/63245456/199061840-068419e4-7032-4529-8a3c-95bfd1d7381d.png)
</details>

<details>
<summary>Commit 2 - adds categories to list and reworks first section</summary>

![Screenshot from 2022-10-31 17-40-54](https://user-images.githubusercontent.com/63245456/199061818-5e63bdb0-4e26-4dee-a8da-1eed6b5c4b03.png)
</details>

timabbott commented:
Merged, thanks @laurynmm!
